src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1075,19 +1084,26 @@ def dbapi(
             snowflake_table_name = random_name_for_temp_object(TempObjectType.TABLE)
             create_table_sql = (
                 "CREATE "
-                + f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
-                + "TABLE "
-                + f"{snowflake_table_name} "
-                + f"""({" , ".join([f'"{field.name}" {convert_sp_to_sf_type(field.datatype)} {"NOT NULL" if not field.nullable else ""}' for field in struct_schema.fields])})"""
+                f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
+                "TABLE "
+                f"{snowflake_table_name} "
+                f"""({" , ".join([f'"{field.name}" {convert_sp_to_sf_type(field.datatype)} {"NOT NULL" if not field.nullable else ""}' for field in struct_schema.fields])})"""
+                f"""{DATA_SOURCE_SQL_COMMENT}"""
+            )
+            self._session.sql(create_table_sql).collect(
+                statement_params=statements_params_for_telemetry
             )
-            self._session.sql(create_table_sql).collect()
             res_df = self.table(snowflake_table_name)
 
             # create temp stage
             snowflake_stage_name = random_name_for_temp_object(TempObjectType.STAGE)
-            sql_create_temp_stage = f"create {get_temp_type_for_object(self._session._use_scoped_temp_objects, True)} stage if not exists {snowflake_stage_name}"
-            self._session._run_query(sql_create_temp_stage, is_ddl_on_temp_object=True)
-
+            sql_create_temp_stage = (
+                f"create {get_temp_type_for_object(self._session._use_scoped_temp_objects, True)} stage"
+                f" if not exists {snowflake_stage_name} {DATA_SOURCE_SQL_COMMENT}"
+            )
+            self._session.sql(sql_create_temp_stage).collect(
+                statement_params=statements_params_for_telemetry
+            )
```

Comment:
can I change to use .sql.collect instead of self._session._run_query?
why those two sql were called differently

Reply:
sure! I actually forget why, but I think this is a safe change